code/utility/logger.cpp

```diff
@@ -11,6 +11,7 @@ void Logger::setStream(QTextEdit* log_stream) {
 
 //TODO: Overload ostream operator
 bool Logger::log (std::string message, LogType type) {
+    if (type == LogType::DEBUG && !Logger::m_debug) return false;

```

Comment:
The logic for whether to print a debug message needs to happen at the caller of the `Logger::log` function, not inside this function.